### ISSUE TITLE
Opentelemetry for scheduler

### ISSUE THREAD

**Issue by Postremus** [quarkusio/quarkus#29364]
### Description

I am using `quarkus-scheduler` and `quarkus-opentelemetry`.
Right now, a scheduler invocation does not create a new span.

```
@Scheduled(every = "1s")
    void schedule() {
        Log.info(Span.current().getSpanContext().getSpanId());
    }
```

This outputs `2022-11-19 17:44:16,006 INFO  [org.acm.Sched] (vert.x-worker-thread-0) 0000000000000000`, which is an invalid span id - no span is open at that point.

Can this please be added?

**Comment by mkouba** [comment]
I wonder if annotating the method with [`@io.opentelemetry.extension.annotations.WithSpan`](https://quarkus.io/guides/opentelemetry#cdi) would work?

**Comment by machi1990** [comment]
I was wondering the same. This issue seems to be related to https://github.com/quarkusio/quarkus/issues/28552

**Comment by brunobat** [comment]
It is the same @machi1990 and using `@WithSpan` on a CDI bean will start a span. However, we should start an Otel context  once a job is started like with in other cases where we are using managed threads... Right @mkouba ?

**Comment by mkouba** [comment]
I believe that since 2.14.0.CR1 we execute the scheduled methods using the managed thread pool (for both `quarkus-scheduler` and `quarkus-quartz`).

@Postremus which version of quarkus do you use?

**Comment by syr** [comment]
> I was wondering the same. This issue seems to be related to #28552

Correct, OTel context init. on quartz job execution is not implemented yet but was moved to the observability roadmap.
https://github.com/quarkusio/quarkus/issues/28552#issuecomment-1277299326

@Postremus 
I worked around the issue for now like: https://github.com/quarkusio/quarkus/issues/28552#issuecomment-1319823367

**Comment by Postremus** [comment]
@mkouba 
> which version of quarkus do you use?
2.14.1.Final

Using `@WithSpan` works, I now get a proper SpanId (like 95d38a815ef4498c).

**Comment by Postremus** [comment]
Closing, since this is solved by #29802
